test/CadlRanchProjectsNonAzure.Tests/model/inheritance/RecursiveTests.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using AutoRest.TestServer.Tests.Infrastructure;
+using NUnit.Framework;
+using Scm._Type.Model.Inheritance.Recursive;
+using Scm._Type.Model.Inheritance.Recursive.Models;
+using Azure.Core;
+using System.Linq;
+
+namespace CadlRanchProjectsNonAzure.Tests
+{
+    public class RecursiveTests : CadlRanchNonAzureTestBase
+    {
+        [Test]
+        public Task Type_Model_Inheritance_Recursive_get() => Test(async (host) =>
+        {
+            var response = await new RecursiveClient(host, null).GetRecursiveAsync();
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual(0, response.Value.Level);
+            Assert.AreEqual(2, response.Value.Extension.Count);
+            Assert.AreEqual(1, response.Value.Extension[0].Level);
+            Assert.AreEqual(1, response.Value.Extension[0].Extension.Count);
+            Assert.AreEqual(2, response.Value.Extension[0].Extension[0].Level);
+            Assert.AreEqual(1, response.Value.Extension[1].Level);
+            Assert.AreEqual(0, response.Value.Extension[0].Extension[0].Extension.Count);
+            Assert.AreEqual(false, !(response.Value.Extension[1].Extension is ChangeTrackingList<Extension> changeTrackingList && changeTrackingList.IsUndefined));
+        });
+        [Test]
+        public Task Type_Model_Inheritance_Recursive_put() => Test(async (host) =>
+        {
+            var data = new Extension(0);
+            var extensions = data.Extension as ChangeTrackingList<Extension>;
+            Extension item1 = new Extension(1);
+            (item1.Extension as ChangeTrackingList<Extension>).Add(new Extension(2));
+            Extension item2 = new Extension(1);
+            extensions.Add(item1);
+            extensions.Add(item2);
```

Comment:
I did not see the purpose of doing this.
We should write the test only using public APIs otherwise we are testing a behavior that our customers could never use.
Could you please rewrite this by removing the cast to `ChangeTrackingList<Extension>`?

Reply:
The purpose of this is to try add more items into list. I think there are some problem related our generated source code.
```
@doc("element")
model Element {
  extension?: Extension[];
}
```
The `Extension` property is optional, but we generated it as `public IReadOnlyList<Extension> Extension { get; }` without setter. It is not correct.

Reply:
Per offline sync up, we have to have this because the usage is not correctly calculated therefore the type is incorrectly generated as `IReadOnlyList`
@mcgallan I believe the latest update on feature/v3 should have fixed this issue, please merge from feature/v3 and the type is this `Extension` property should be `IList` now, and we could properly add element into it

Reply:
fixed